### PR TITLE
Doubletap

### PR DESCRIPTION
**Pull request by DoomFishWasTaken**
Doubletap made by @notgoodusename, edited and made compatible with the newest Osiris version by me

### CODE REVIEW

**Comment by yaelahrip** [comment]
@DoomFishWasTaken can i have your  discord  buddy  ? wanna  ask something

**Comment by SapphireCodenz** [comment]
> @DoomFishWasTaken can i have your discord buddy ? wanna ask something

DoomFish#1337

**Comment by SwagsRGB** [comment]
please tell me what is double tap? i am noob sorry

**Comment by DoomFishWasTaken** [comment]
> please tell me what is double tap? i am noob sorry

Exploit clientside simulation to shoot two (or more) bullets per shot

**Review by yaelahrip — COMMENTED** [review]
> 
> 
> > @DoomFishWasTaken can i have your discord buddy ? wanna ask something
> 
> DoomFish#1337


got  replied by different account -_-  are u doom  ?

**Comment by gr1ndy** [comment]
Um.. i got 1821 errors after building :| add me on discord i need help
SpraYer#9710

**Comment by dayz2000** [comment]
it's legit feature?

**Comment by tirziz** [comment]
> it's legit feature?

No

**Comment by Ferryistaken** [comment]
I think that it would be a nice feature to add but osiris wouldn't really benefit from it. The devs should probably create a ragebot tab and let the community fill that up

File: Osiris/Hacks/Tickbase.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    Tickbase::tick->ticksAllowedForProcessing = Tickbase::tick->maxUsercmdProcessticks - Tickbase::tick->chokedPackets;
+    Tickbase::tick->ticksAllowedForProcessing = std::clamp(Tickbase::tick->ticksAllowedForProcessing, 0, Tickbase::tick->maxUsercmdProcessticks);
+}
+
+void Tickbase::shiftTicks(int ticks, UserCmd* cmd, bool shiftAnyways) noexcept //useful, for other funcs
+{
+    if (!localPlayer || !localPlayer->isAlive() || !config->ragebotExtra.enabled)
+        return;
+
+    if (!canShift(ticks, shiftAnyways))
```
[the remark this answers is not in the document]
Comment (DoomFishWasTaken) [reply Osiris/Hacks/Tickbase.cpp:59]: just some old comments ignore em

**Comment by pparadoxx** [comment]
> Um.. i got 1821 errors after building :| add me on discord i need help
> SpraYer#9710

you prob don't have the libs required for building things, to fix this u just need to update

**Comment by degeneratehyperbola** [comment]
getUserCmd(slot, from) always returns a null pointer. I can't seem to figure out why.

**Pull request closed without merging**